tests/storage/test_event_federation.py

```diff
@@ -506,11 +508,24 @@ def insert_event(txn):
         )
         self.assertSetEqual(difference, set())
 
-    def test_prune_inbound_federation_queue(self):
-        "Test that pruning of inbound federation queues work"
+    @parameterized.expand(
+        [
+            (RoomVersions.V1,),
+            (RoomVersions.V9,),
+        ]
+    )
+    def test_prune_inbound_federation_queue(self, room_version: RoomVersion):
```

Comment:
ooi have you tested that this test fails without the patch?

Reply:
yep, it fails.